Rebrand and update lightstep docs

robswhitmore opened the pull request:
Please provide a description for what this PR is for.

Lightstep has been rebranded to Cloud Observability from ServiceNow. Additionally, we now use the OpenTelemetry Collector to ingest metrics, traces, and logs from third-parties. These updates include both changes.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

- [ ] Ambient
- [ X ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Extensions and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

istio-testing commented:
Hi @robswhitmore. Thanks for your PR.

I'm waiting for a [istio](https://github.com/orgs/istio/people) member to verify that this patch is reasonable to test. If it is, they should reply with `/ok-to-test` on its own line. Until that is done, I will not automatically test new commits in this PR, but the usual testing commands by org members will still work. Regular contributors should [join the org](https://github.com/orgs/istio/people) to skip this step.

Once the patch is verified, the new status will be reflected by the `ok-to-test` label.

I understand the commands that are listed [here](https://go.k8s.io/bot-commands?repo=istio%2Fistio.io).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

robswhitmore commented:
I'd like to also add the label `cherrypick/release-1.19` but I don't seem to be able to edit the labels field.

ericvn commented:
/ok-to-test

ericvn commented:
Depending on when this merges, it may or may not need the 1.19 label as there is no 1.19 branch (right now). The branch will be created when 1.19.0 is release.d.

istio-testing commented:
@robswhitmore: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
lint_istio.io | b7a080300104a63772ef391da6ba94c613223df5 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/13779/lint_istio.io/1696311851547103232) | true | `/test lint`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

Arhell reviewed (commented):
/ok-to-test

istio-testing commented:
@robswhitmore: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
lint_istio.io | b7a080300104a63772ef391da6ba94c613223df5 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/13779/lint_istio.io/1696323908967337984) | true | `/test lint`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

robswhitmore commented:
I fixed the lint errors that I can, but the errors for spacing around lists can't be fixed because those are inside yaml code blocks. What should I do?

Also, what should we tag this PR so that it goes into the latest release?

istio-testing commented:
@robswhitmore: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
lint_istio.io | 13039b79aa8c009b1e627a172a2afe12f25112b6 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/13779/lint_istio.io/1696534025302183936) | true | `/test lint`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

ericvn commented:
@istio/wg-policies-and-telemetry-maintainers Can you review this as owners of the page as well. Thanks.

ericvn commented on content/en/docs/tasks/observability/distributed-tracing/lightstep/index.md:
```diff
@@ -1,197 +1,237 @@
     {{< /text >}}
 
-### Deploy Istio with Public or Developer Mode Satellites
+2. Create an `otel-collector-deployment.yaml` file by copying the following code.
```
```suggestion
1. Create an `otel-collector-deployment.yaml` file by copying the following code.
```

ericvn commented on content/en/docs/tasks/observability/distributed-tracing/lightstep/index.md:
```diff
@@ -1,197 +1,237 @@
+    {{< /text >}}
 
-1.  Select `productpage.default` from the **Service** drop-down list.
+2. Apply the ConfigMap to your Kubernetes cluster
```
```suggestion
1. Apply the ConfigMap to your Kubernetes cluster
```

ericvn commented on content/en/docs/tasks/observability/distributed-tracing/lightstep/index.md:
```diff
@@ -1,197 +1,237 @@
+    {{< /text >}}
 
-    {{< image link="./istio-tracing-details-lightstep.png" caption="Detailed Trace View" >}}
+3. Apply the Secret to your Kubernetes cluster
```
```suggestion
1. Apply the Secret to your Kubernetes cluster
```

ericvn reviewed (commented):
I believe these 3 suggestions will fix the numbering. Looks like the line numbering is off by 1.

lei-tang commented on content/en/docs/tasks/observability/distributed-tracing/lightstep/index.md:
```diff
@@ -1,197 +1,237 @@
-that the call took 15.30 ms. The second span is labeled with the `reviews.default.svc.cluster.local:9080/*` operation
-and the `reviews.default: proxy server` service. The second span is a child of the first span and represents the
-server-side span of the call. The screenshot shows that the call took 14.60 ms.
+## View metrics in Cloud Observability
```
The folder content/en/docs/tasks/observability/distributed-tracing/lightstep is for tracing. Consider placing the metric guide into a different folder.

robswhitmore replied:
Moved to the metrics section

lei-tang commented on content/en/docs/tasks/observability/distributed-tracing/lightstep/index.md:
```diff
@@ -1,197 +1,237 @@
 ---
 
-{{< boilerplate telemetry-tracing-tips >}}
+You use the [OpenTelemetry Collector](https://opentelemetry.io/docs/collector/) to send Istio metrics to Cloud Observability using Istio proxies within each pod. Metrics are collected and forwarded to the OpenTelemetry Collector, which acts as a central collection and processing point and then sends those metrics to Cloud Observability.
```
The folder content/en/docs/tasks/observability/distributed-tracing/lightstep is for tracing. Consider placing the metric guide into a different folder.

robswhitmore replied:
Moved to the metrics section

robswhitmore replied:
Removed references to Lightstep in site where it was mentioned in the context of tracing (this also fixed broken links caused when moving the topics from Tracing to Metrics). There is still a broken link in `/news/releases/1.1.x/announcing-1.1/change-notes/`, but I don't think I should change that, right?

istio-testing commented:
@robswhitmore: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
lint_istio.io | 474cebce3d1aa5346704a69a3ed5686ca3ecfa9b | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/13779/lint_istio.io/1696594569279311872) | true | `/test lint`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@robswhitmore: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
lint_istio.io | 9c381e7aab5a37ce0b7cf8ca8b49f579683b2c0b | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/13779/lint_istio.io/1696657866649243648) | true | `/test lint`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

robswhitmore commented:
/test lint

istio-testing commented:
@robswhitmore: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
lint_istio.io | 5aaeaf6b0cd433074cfd328a36dc04ece98507c3 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/13779/lint_istio.io/1696667268362014720) | true | `/test lint`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

robswhitmore commented:
I don't think I can fix this link because it's in a news post, right?

robswhitmore commented:
@lei-tang or whoever else might be able to help -
The linter is failing with a broken link to an old change note. I don't think I should update this, correct?
![image](https://github.com/istio/istio.io/assets/50384513/ad53cb7a-05a2-425f-a492-ce944933dbdb)

lei-tang commented:
@ericvn Do you have suggestions for fixing the broken link error above?

ericvn commented on content/en/docs/tasks/observability/metrics/lightstep/index.md:
```diff
@@ -0,0 +1,232 @@
+weight: 11
+keywords: [telemetry, metrics ,lightstep, servicenow, cloud observability]
+aliases:
+ - /docs/tasks/observability/metrics/lightstep/
```
Why do you have this alias here? It's just pointing to itself.

ericvn commented:
Lint error is Running linkinator...
  [404] http://localhost:5080/docs/tasks/observability/distributed-tracing/lightstep/
ERROR: Detected 1 broken links. Scanned 5787 links in 39.311 seconds.

You removed this file so makes sense that the link isn't found.

What do you want people who pointed to it to point to now?  You could point to an older releases version of the file by changing the URL to point to a specific older release.  Maybe you want to add an alias to a new page so the page would then be used in the other pages place (ie. replace the alias I've asked you to remove with `/docs/tasks/observability/distributed-tracing/lightstep/`).

ericvn commented:
Also added the cherry-pick label since we have cut the new branch.

istio-testing commented:
@robswhitmore: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
lint_istio.io | d5c1904037564fa2c2bdae0934a94fe3f81cefd9 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/13779/lint_istio.io/1699554919234670592) | true | `/test lint`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

robswhitmore commented:
@ericvn I did what you suggested and changed the alias to the page I deleted (/docs/tasks/observability/distributed-tracing/lightstep/), but I'm still getting the same broken link and now an additional broken link for that alias. I followed the directions [here](/docs/tasks/observability/distributed-tracing/lightstep/), but I must be missing something?

dhawton commented:
> @ericvn I did what you suggested and changed the alias to the page I deleted (/docs/tasks/observability/distributed-tracing/lightstep/), but I'm still getting the same broken link and now an additional broken link for that alias. I followed the directions [here](/docs/tasks/observability/distributed-tracing/lightstep/), but I must be missing something?

The lint report tells you the file the link is originating from, which is https://github.com/robswhitmore/istio.io/blob/robswhitmore/rebrand-update-lightstep/content/en/news/releases/1.1.x/announcing-1.1/change-notes/index.md?plain=1#L165

Since you're changing the name of the file, link to an archived version of the page, for example: https://istio.io/v1.18/docs/tasks/observability/distributed-tracing/lightstep/

ericvn commented:
@robswhitmore Daniel is correct. There are two ways to fix the issue. One is to update the link which is pointing to the file you removed in `news/releases/1.1.x/announcing-1.1/change-notes/` to point to and older existing version of the page (use 1.19 or 1.18 for example). You can look at a commit where I did that yesterday: https://github.com/istio/istio.io/commit/fbb5c8df9bf9f0e6969a4413f0a669c69b62042b prefixing `https://archive.istio.io/v1.17` to the /docs/... that was removed. In this case, the link will point to the same information as before.

As I noted, if the new page you replaced the deleted page with is an acceptable alternative, then you can simply add `[/docs/tasks/observability/distributed-tracing/lightstep/](https://istio.io/v1.18/docs/tasks/observability/distributed-tracing/lightstep/)` to the aliases list on the new page, and when they click the link it will take them to the new page.

Pick whichever way yields the best information to the reader.

ericvn commented on content/en/docs/tasks/observability/metrics/lightstep/index.md:
```diff
@@ -0,0 +1,232 @@
+weight: 11
+keywords: [telemetry, metrics ,lightstep, servicenow, cloud observability]
+aliases:
+ - /docs/tasks/observability/distributed-tracing/lightstep/
```
I'm wondering if there is a format issue with number of leading spaces, as this does look correct. Need to try pulling your PR and trying the link-check locally.
```suggestion
    - /docs/tasks/observability/distributed-tracing/lightstep/
```

ericvn replied:
Basically, the alias discussion is here: https://istio.io/latest/docs/releases/contribute/front-matter/#rename-move-or-delete-pages

ericvn replied:
I did pull your branch locally and try a few things with the aliases, and I never got it to work. I guess I would just do option 1 and update the 1 link to point to the archive and remove the alias.

ericvn replied:
@frankbu is there some magic on the `aliases:`?

ericvn commented on content/en/docs/tasks/observability/metrics/lightstep/index.md:
```diff
@@ -0,0 +1,232 @@
+---
+title: Viewing Istio Metrics in Cloud Observability from ServiceNow
+description: This task shows you how to use OpenTelemetry to ingest and view Istio metrics in Cloud Observability (formerly Lightstep).
+weight: 11
```
Also the weight here is used to set the order of the links within the page. You can look at your preview at https://deploy-preview-13779--preliminary-istio.netlify.app/latest/docs/tasks/observability/metrics/ to see if the orderings correct. Higher numbers go down the order.

robswhitmore commented:
@ericvn and @lei-tang - my commit finally passed the lint check, so I think it's good to go. Thanks for all your help!

lei-tang commented on content/en/docs/tasks/observability/metrics/lightstep/index.md:
```diff
@@ -0,0 +1,230 @@
+
+## Configure Istio Operator
+
+Create an `istio-operator.yaml` file by copying the following code.
```
Is 'istio-operator.yaml' here used in the guide? A quick search did not return any results.

robswhitmore replied:
I'm not sure what you mean by "used in the guide?" Do you mean elsewhere in Istio docs? It's used in this topic to configure the k8s operator for Istio.

lei-tang replied:
> "It's used in this topic to configure the k8s operator for Istio."
Which command lines in this guide use 'istio-operator.yaml'?

robswhitmore replied:
Thanks - I realized what he was asking and agreed, so I sent it back to the engineering team. They are working on updates.Thanks for catching that - I definitely should have, but with this rebrand, I've been 🤪!

ericvn commented:
Does https://github.com/istio/istio.io/pull/13996 replace this?

robswhitmore commented:
News to me! 😕  Let me find out what's going on...

robswhitmore commented:
Looks like the eng team decided to focus on Tracing, so that new PR is separate from this one, and is taking precedence. Sorry for all the confusion!

ericvn commented:
> Looks like the eng team decided to focus on Tracing, so that new PR is separate from this one, and is taking precedence. Sorry for all the confusion!

I think you are saying that at some point this PR will be revisited.

robswhitmore commented:
Yes, at some point. Thanks -

istio-testing commented:
PR needs rebase.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

craigbox commented:
We've decided to remove this from the Istio docs, as Lightstep just takes traces in OTLP format and there's nothing specific to add above and beyond that.

We'll link https://docs.lightstep.com/integrations/ingest-metrics-istio from the OTLP docs to make it clear what people need to do.
